src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -570,6 +569,26 @@ CreateShardsOnWorkersViaExecutor(Oid distributedRelationId, List *shardPlacement
 		taskList = lappend(taskList, task);
 	}
 
+	if (useExclusiveConnection)
+	{
+		/*
+		 * When the table has local data, we force max parallelization so data
```

Comment:
I think we're doing it in two cases: local table has data or we're inside a transaction block.

So, maybe add `We also prefer to use max parallelization when we're inside a transaction block because the user might execute compute heavy commands later in the transaction block.`